hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -24,199 +24,356 @@
       tests: []
     }
     {
-      name: intr_enabled
+      name: fifo_eviction
       desc: '''
-            Sanity + enable all interrupts and verify correctness of assertion and de-assertion of
-            each pin. Replace polling with waits on interrrupts.
+            Perform following sequences of operations: read/program/read and read/erase/read in
+            order to test fifo eviction properly. Read should be executed by both Software and
+            Host interface. All combinations should be tested. Covergroup for this hazardous
+            behavior is fifo_evict_cg.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: host_ctrl_arb
+      name: host_arb
       desc: '''
-            Continuously read data from host interface while randomly programming or reading from
-            the flash controller. Ensure the expected bandwidths are seen on both interfaces. Also,
-            initiate reads from the host to all banks back and forth to ensure the responses arrive
-            in-order.
+            Test arbitration within Flash Physical Controller by reading from both interfaces at
+            the same time. Perform continuously direct read data from host interface and at the
+            same time, perform all operations READ/PROGRAM/ERASE from the flash controller is in
+            progress. Perform parallel operations at addresses of different banks and also on same
+            bank. Expect that operations are successfully executed.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: host_read_collide
+      name: host_interleave
       desc: '''
-            Read LSB bus word via host and program or read the MSB bus word at the same time. This
-            will cause the internal flash word wide buffer to be updated. Then read the MSB bus word
-            within the same flash data word - it should read back the updated data, proving that the
-            internal buffer was cleared from the previous step.
+            At same time, perform two read operations and the same time via hardware host and via
+            controller. At same time, perform read operation via hardware host and program
+            operation via controller. Perform mentioned parallel operations at different addresses
+            and on the same address (i.e. seed material). Expect that operations are successfully
+            executed.
             '''
       milestone: V2
       tests: []
     }
     {
-      name: en_mp_regions
+      name: memory_protection
       desc: '''
-            Protect randomized and legal ranges of data and info partitions and ensure that the
-            corresponding read / program / erase enable bits are working. TBD - future support for
-            sub-page protection may be added - that will be tracked as a part of this test as well.
+            Perform READ/ PROGRAM/ ERASE operations over protected regions and pages of data and
```

Comment:
just a note here that there is a specific register used to protect mass erase, please make sure that's included as well.

Reply:
This is explained in second sentence " Use set and reset values of corresponding read, program and erase enable bits.".